src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -11924,6 +11924,54 @@ HRESULT CEEJitInfo::getMethodBlockCounts (
     return hr;
 }
 
+CORINFO_CLASS_HANDLE CEEJitInfo::getLikelyClass(
+                     CORINFO_METHOD_HANDLE ftnHnd,
+                     CORINFO_CLASS_HANDLE  baseHnd,
+                     UINT32                ilOffset,
+                     UINT32 *              pLikelihood,
+                     UINT32 *              pNumberOfClasses
```

Comment:
pLikelyhood couls also be a 32-bit float.
As it currently stands it is an integer percentage, where 100 means 100% and 33 means 33%

Reply:
Same goes here, histogram data is approximate.

What mostly matters is that for monomorphic types we get it right and predict high likelihood. For other cases the payoff is smaller.